apis/getPatientDemographics/index.js

```diff
@@ -0,0 +1,55 @@
+/*
+
+ ----------------------------------------------------------------------------
+ |                                                                          |
+ | Copyright (c) 2019 Ripple Foundation Community Interest Company          |
+ | All rights reserved.                                                     |
+ |                                                                          |
+ | http://rippleosi.org                                                     |
+ | Email: code.custodian@rippleosi.org                                      |
+ |                                                                          |
+ | Author: Rob Tweed, M/Gateway Developments Ltd                            |
+ |                                                                          |
+ | Licensed under the Apache License, Version 2.0 (the "License");          |
+ | you may not use this file except in compliance with the License.         |
+ | You may obtain a copy of the License at                                  |
+ |                                                                          |
+ |     http://www.apache.org/licenses/LICENSE-2.0                           |
+ |                                                                          |
+ | Unless required by applicable law or agreed to in writing, software      |
+ | distributed under the License is distributed on an "AS IS" BASIS,        |
+ | WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. |
+ | See the License for the specific language governing permissions and      |
+ |  limitations under the License.                                          |
+ ----------------------------------------------------------------------------
+
+  14 March 2019
+
+*/
+
+'use strict';
+
+const { logger } = require('../../lib/core');
+const { GetDemographicsCommand } = require('../../lib/commands');
+const { getResponseError } = require('../../lib/errors');
+
+/**
+ * @param  {Object} args
+ * @param  {Function} finished
+ */
+module.exports = async function getPatientDemographics (args, finished) {
+
+  try {
+    const command = new GetDemographicsCommand(args.req.ctx, args.session);
+    const responseObj = await command.execute(args.patientId);
+    
+    finished(responseObj);
+  } catch (err) {
+    
+    logger.error(err);
```

Comment:
@richardcbrown Please check this one
https://github.com/killmenot/QEWD-Courier/blob/patch-1/helm/openehr_service/apis/getFeedsSummary/index.js#L49

passing string as first arg allows to know the specific place where error is occurred (but you have one handler here)